### DISCUSSION TITLE
scale, prevent outlier, remove process noise for speedup

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#124] [people; relating]
For the first topic (speedup) from https://github.com/Data4DM/BayesSD/discussions/76#discussioncomment-4269232 hier.ode.sbc blueprint, scaling, removing process noise structure, removing data series structure are tried

## scaling
parameters are tight enough, but additive measurement noise is larger than other parameter as initial order 10000. 
Q1. Will scaling this to 0~1 would make the fitting better in both accuracy and speed.

Long fitting time accompanies the following log:
> Exception: normal_lpdf: Location parameter[1] is nan, but must be finite! (in '/Users/hyunjimoon/Dropbox/stanify/vignette/stan_files/Inven_S20N10R2_M200/data2draws.stan', line 73, column 8 to column 97)

```
prior = {
    ("inventory_adjustment_time", "normal", 8.0, 0.8, 0),
    ("wip_adjustment_time", "normal", 2, 0.02, 0),
    ("m_noise_scale", "normal", init_order * .1, init_order * .01, 0)
}
```
<img width="558" alt="image" src="https://user-images.githubusercontent.com/30194633/206701724-31a7d75e-2876-4337-8773-176302d5ce86.png">

Q2. will sd people ready to actually sample from prior distribution for baseline run? Most modeler's mental model seems to be using it as a penalty function for estimation

## thin tail
either by `m_scale_sd` shape or measurement distribution. inven model time log: 
```
turn off pnoise, normal m_noise_scale: 1.38min
turn off pnoise, normal m_noise_scale, N:200: 5min
turn off pnoise, lognormal m_noise_scale, : 33 min
```

Q3. is it correct the second one is "likelihood to lognormal" formulation? what is the main difference between the two?
```
model{
    wip_adjustment_time ~ normal(2, 0.02);
    m_noise_scale ~ lognormal(10.0, 1.0);
    inventory_adjustment_time ~ normal(rep_vector(8.0, R), 0.8);
    for (r in 1:R){
        production_rate_stocked_obs ~ normal(production_rate_stocked, m_noise_scale);
        production_start_rate_stocked_obs ~ normal(production_start_rate_stocked, m_noise_scale);
     }
}
```
vs
```
model{
    wip_adjustment_time ~ normal(2, 0.02);
    m_noise_scale ~ normal(10.0, 1.0);
    inventory_adjustment_time ~ normal(rep_vector(8.0, R), 0.8);
    for (r in 1:R){
        production_rate_stocked_obs ~ lognormal(production_rate_stocked, m_noise_scale);
        production_start_rate_stocked_obs ~ lognormal(production_start_rate_stocked, m_noise_scale);
     }
}
```

## process noise structure
is removed for 12W1 experiments on inventory model but the data series is still driving data (changed to sin wave)

**Comment by tomfid** [comment]
Q2. I think people would readily accept sampling from the prior to generate a baseline. This is consistent with what people already do with multivariate sensitivity runs, even in data-free models. I think it needs to be easier though. People already underuse sensitivity analysis, because it adds an extra set of time-consuming tasks to the modeling process. So the difficult behavior change is to reallocate the distribution of time spent on the entire package.

**Reply by tomfid** [reply]
Q3. I'm not sure about the "likelihood to lognormal" phrase, but the difference (if I'm reading this right) is that Normal error (a) makes the meas error insensitive to the scale of the measured variable, (b) potentially admits negative values, though this only occurs if the SD is large enough relative to the underlying variable, and (c) the probability is symmetric around the true value in variable-space. The LogNormal otoh (a) makes the error proportional to scale, (b) is zero-avoiding, and (c) is symmetric in log-variable-space, but skewed right (heavy tail) in variable-space.

When you switch to the lognormal distribution, are you changing the prior accordingly? Seems like the Normal

("m_noise_scale", "normal", init_order * .1, init_order * .01, 0)

Should become something like:

("m_noise_scale", "lognormal",  .1,  .01, 0)

because it now represents a geometric standard deviation rather than an absolute sd. If not, this might explain the long run time.

**Comment by hyunjimoon** [comment]
Increasing N:10 -> 100 -> 200, fitting time increase 1-> 15s -> 35s  (slight superlinear) 
because of stan's following error`fatal error: bracket nesting level exceeded maximum of 256`, N < 256 with current stanfity structure.
